eox_nelp/processors/xapi/mixins.py

```diff
@@ -0,0 +1,47 @@
+"""Mixin file, this is a collection of classes that could be reused in multiple implementations.
+
+Classes:
+    - BaseCourseObjectMixin: Base mixin for transformers that has as object an edx-platform course
+"""
+from django.utils.functional import cached_property
+from tincan import Activity, ActivityDefinition, LanguageMap
+
+from eox_nelp.processors.xapi import constants as eox_nelp_constants
+from eox_nelp.utils import get_course_from_id
+
+
+class BaseCourseObjectMixin:
+    """
+    Base mixin for transformers that has as object an edx-platform course.
+    """
+    @cached_property
+    def course_id(self):
+        """The course id data depends on the kind of event, an event could publish this as course_id or course_key
```

Comment:
Interesting but not standardized. I understand the thing, but this is not our fault. So yes the class is a solution to change the way to get the course_id

Reply:
This could be wider, let's think in a event that shares the usage key so the course id has to be extracted or calculated